lib/src/custom_sentence.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:nmea/src/checksum_sentence.dart';
+import 'package:nmea/src/nmea_sentence.dart';
+import 'package:nmea/src/nmea_sentence_type.dart';
+
+/// The data format is completely customizable, as long as it fits in the general
+/// envelope of an NMEA0183 sentence (begins with '$' and ends with windows-
+/// style newline characters).
+class CustomSentence extends ChecksumSentence {
```

Comment:
I'd like to support sentences that could have checksums as well. Can I maybe add a parameter to ChecksumSentence to enforce strictness?

Reply:
Hmmm... Maybe we can rework this into an extension class?

```dart
extension ChecksumExtension on NmeaSentence { ... }
```

...instead of having an abstract class for all checksum-enabled sentences.

For now, I suggest that `CustomSentence` extends `NmeaSentence` and you implement the checksum in your own implementations (i.e. not on library level).

Reply:
Problem with that approach is that the last field value is read with the checksum and the checksum isn't parsed from the field value

Reply:
Currently the caller can define whether they want the decoder to allow invalid sentences that don't have a valid checksum. I think custom sentences should extend from checksum sentence so that it can cater for checksums when they do exist and gracefully fail if they don't based on the decoder configuration.

Reply:
I agree this would be a possible solution.
Another solution may be to add another option to the decoder to disable checksum checks for custom sentences.
Further, we could move this flag to the `CustomSentence` itself, so the implementer can decide if the sentence uses a checksum and if it should be validated.

What do you think?

Reply:
Yeah that's what I suggested earlier. Let me try and you can see if you ware happy with what I came up with?

Reply:
Pushed an update with the new parameter

Reply:
Looks good! Please note my comment down below and make sure you make a `dart format lib` before committing.